text/0044-rent-based-registration.md

```diff
@@ -0,0 +1,201 @@
+
+## Requirements
+
+1. The solution SHOULD NOT affect the current model for registering validation code.
+2. The solution SHOULD offer an easily configurable way for governance to adjust the initial deposit and recurring rent cost.
+3. The solution SHOULD provide an incentive to prune validation code for which rent is not paid.
+4. The solution SHOULD allow anyone to re-register validation code under the same `ParaId` without the need for redundant pre-checking if it was already verified before.
+5. The solution MUST be compatible with the Agile Coretime model, as described in RFC#0001
+6. The solution MUST allow anyone to pay the rent.
+7. The solution MUST prevent the removal of validation code if it could still be required for disputes or approval checking.
+
+## Stakeholders
+
+- Future Polkadot on-demand Parachains
+
+## Explanation
+
+This RFC proposes a set of changes that will enable the new rent based approach to registering and storing validation code on-chain. 
+The new model, compared to the current one, will require periodic rent payments. The parachain won't be pruned automatically if the rent is not paid, but by permitting anyone to prune the parachain and rewarding the caller, there will be an incentive for the removal of the validation code.
+
+On-demand parachains should still be able to utilize the current one-time payment model. However, given the size of the deposit required, it's highly likely that most on-demand parachains will opt for the new rent-based model.
+
+Importantly, this solution doesn't require any storage migrations in the current system nor does it introduce any breaking changes. The following provides a detailed description of this solution.
+
+### Registering an on-demand parachain
+
+In the current implementation of the registrar pallet, there are two constants that specify the necessary deposit for parachains to register and store their validation code:
+```rust
+trait Config {
+	// -- snip --
+
+	/// The deposit required for reserving a `ParaId`.
+	#[pallet::constant]
+	type ParaDeposit: Get<BalanceOf<Self>>;
+
+	/// The deposit to be paid per byte stored on chain.
+	#[pallet::constant]
+	type DataDepositPerByte: Get<BalanceOf<Self>>;
+}
+```
+
+This RFC proposes the addition of three new constants that will determine the payment amount and the frequency of the recurring rent payment:
+```rust
+trait Config {
+	// -- snip --
+
+	/// Defines how frequently the rent needs to be paid.
+	///
+	/// The duration is set in sessions instead of block numbers.
+	#[pallet::constant]
+	type RentDuration: Get<SessionIndex>;
+
+	/// The initial deposit amount for registering validation code.
+	///
+	/// This is defined as a percentage of the deposit that would be required in the regular
+	/// model.
+	#[pallet::constant]
+	type InitialRentDeposit: Get<Perbill>;
```

Comment:
bikeshed: given that these are proportions, how about `RentalDepositProportion` and `RentalRecurringProportion`?